src/datetime_queries/utils.rs

```diff
@@ -0,0 +1,81 @@
+use chrono::{DateTime, Datelike, Duration, NaiveDate, Timelike, Utc};
+use hdk::prelude::*;
+
+pub fn is_valid_date_range(
+    start: FetchEntriesTime,
+    end: FetchEntriesTime,
+) -> Result<(), WasmError> {
+    match start.to_date_time() < end.to_date_time() {
+        true => Ok(()),
+        false => Err(err("invalid date range")),
+    }
+}
+pub fn next_day(date_time: DateTime<Utc>) -> DateTime<Utc> {
+    let next_day = date_time + Duration::days(1);
+    DateTime::from_utc(
+        NaiveDate::from_ymd(next_day.year(), next_day.month(), next_day.day()).and_hms(0, 0, 0),
+        Utc,
+    )
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct FetchEntriesTime {
+    pub year: i32,
+    pub month: u32,
+    pub day: u32,
+    pub hour: Option<u32>,
+}
```

Comment:
This being less of a util, better to move it out of here. Could have a standalone file called `inputs` that holds this